audio.go

```diff
@@ -0,0 +1,164 @@
+	// the temperature until certain thresholds are hit.
+	Temperature float32
+}
+
+type TranscribeOptions struct {
+	*AudioOptions
+	// The language of the input audio. Supplying the input language in ISO-639-1
+	// format will improve accuracy and latency.
+	Language string
+}
+
+type TranscribeResponse struct {
+	Text string `json:"text"`
+}
+
+// Transcribe audio into the input language.
+//
+// Docs: https://platform.openai.com/docs/api-reference/audio/create
+func (e *Engine) Transcribe(ctx context.Context, options *TranscribeOptions) (*TranscribeResponse, error) {
+	if err := e.validate.StructCtx(ctx, options); err != nil {
+		return nil, err
+	}
+
+	url := e.apiBaseURL + "/audio/transcriptions"
+
+	body, contentType, err := newTranscribeBody(options)
+	if err != nil {
+		return nil, err
+	}
+	req, err := e.newReq(ctx, "POST", url, contentType, body)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := e.doReq(req)
+	if err != nil {
+		return nil, err
+	}
+	var jsonResp TranscribeResponse
+	if err := unmarshal(resp, &jsonResp); err != nil {
+		return nil, err
+	}
+	return &jsonResp, nil
+}
+
+func newTranscribeBody(options *TranscribeOptions) (io.Reader, string, error) {
+	body := &bytes.Buffer{}
+	writer, err := newMultiPartWriter(body, options.AudioOptions)
+	if err != nil {
+		return nil, "", err
+	}
+	if options.Language != "" {
+		if err := writer.WriteField("language", options.Language); err != nil {
+			return nil, "", fmt.Errorf("write language: %w", err)
+		}
+	}
+	if err := writer.Close(); err != nil {
+		return nil, "", fmt.Errorf("close writer: %w", err)
+	}
+	return body, writer.FormDataContentType(), nil
+}
+
+type TranslateOptions struct {
+	*AudioOptions
+}
+
+type TranslateResponse struct {
+	Text string `json:"text"`
+}
+
+// Translate audio into English.
+//
+// Docs: https://platform.openai.com/docs/api-reference/audio/create
+func (e *Engine) Translate(ctx context.Context, options *TranslateOptions) (*TranslateResponse, error) {
+	if err := e.validate.StructCtx(ctx, options); err != nil {
+		return nil, err
+	}
+
+	url := e.apiBaseURL + "/audio/translations"
+
+	body, contentType, err := newTranslateBody(options)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := e.newReq(ctx, "POST", url, contentType, body)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := e.doReq(req)
+	if err != nil {
+		return nil, err
+	}
+	var jsonResp TranslateResponse
+	if err := unmarshal(resp, &jsonResp); err != nil {
+		return nil, err
+	}
+	return &jsonResp, nil
+}
+
+func newTranslateBody(options *TranslateOptions) (io.Reader, string, error) {
+	body := &bytes.Buffer{}
+	writer, err := newMultiPartWriter(body, options.AudioOptions)
+	if err != nil {
+		return nil, "", err
+	}
+	if err := writer.Close(); err != nil {
+		return nil, "", fmt.Errorf("close writer: %w", err)
+	}
+	return body, writer.FormDataContentType(), nil
+}
+
+func newMultiPartWriter(body *bytes.Buffer, options *AudioOptions) (*multipart.Writer, error) {
```

Comment:
This multipart writer depends on `AudioOptions`, change name to `newAudioMultipartWriter`